From d1e3108060b3417161e624e3f9390190aba7d9c6 Mon Sep 17 00:00:00 2001
From: Sam Yang <r52096@freescale.com>
Date: Mon, 31 Mar 2008 10:16:17 +0800
Subject: [PATCH] ENGR00070460 Fix FM config HW event macro and deplicated I2C register

1. Add FM item in MX31 3DS and MX37 3DS default config.
2. Fix wrong HW event Macro in pmic_battery.c
MXC_HWEVENT
To
CONFIG_MXC_HWEVENT
3. Delete duplicated I2C register code in mx3_3stack.c

Signed-off-by: Sam Yang <r52096@freescale.com>
---
 arch/arm/configs/imx31_3stack_defconfig |    6 +++---
 arch/arm/configs/imx37_3stack_defconfig |    4 +++-
 arch/arm/mach-mx3/mx3_3stack.c          |    3 ---
 drivers/char/Kconfig                    |    2 ++
 drivers/mxc/pmic/mc13783/pmic_battery.c |   12 ++++++------
 5 files changed, 14 insertions(+), 13 deletions(-)

diff --git a/arch/arm/configs/imx31_3stack_defconfig b/arch/arm/configs/imx31_3stack_defconfig
index 821fa63..6d1abbf 100644
--- a/arch/arm/configs/imx31_3stack_defconfig
+++ b/arch/arm/configs/imx31_3stack_defconfig
@@ -1,7 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Linux kernel version: 2.6.24
-# Thu Mar 20 14:44:17 2008
+# Mon Mar 31 09:56:31 2008
 #
 CONFIG_ARM=y
 CONFIG_SYS_SUPPORTS_APM_EMULATION=y
@@ -529,7 +529,7 @@ CONFIG_BLK_DEV_IDE_MXC=m
 # CONFIG_BLK_DEV_HT6560B is not set
 # CONFIG_BLK_DEV_QD65XX is not set
 # CONFIG_BLK_DEV_UMC8672 is not set
-# CONFIG_BLK_DEV_IDEDMA is not set
+CONFIG_BLK_DEV_IDEDMA=y
 CONFIG_IDE_ARCH_OBSOLETE_INIT=y
 # CONFIG_BLK_DEV_HD is not set
 
@@ -590,7 +590,6 @@ CONFIG_SCSI_LOWLEVEL=y
 CONFIG_ATA=m
 # CONFIG_ATA_NONSTANDARD is not set
 # CONFIG_PATA_LEGACY is not set
-# CONFIG_PATA_PCMCIA is not set
 # CONFIG_PATA_QDI is not set
 # CONFIG_PATA_WINBOND_VLB is not set
 # CONFIG_PATA_PLATFORM is not set
@@ -714,6 +713,7 @@ CONFIG_VT_CONSOLE=y
 CONFIG_HW_CONSOLE=y
 # CONFIG_VT_HW_CONSOLE_BINDING is not set
 # CONFIG_SERIAL_NONSTANDARD is not set
+CONFIG_FM_SI4702=m
 
 #
 # Serial drivers
diff --git a/arch/arm/configs/imx37_3stack_defconfig b/arch/arm/configs/imx37_3stack_defconfig
index 618a043..844cf6b 100644
--- a/arch/arm/configs/imx37_3stack_defconfig
+++ b/arch/arm/configs/imx37_3stack_defconfig
@@ -1,6 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Linux kernel version: 2.6.24
+# Mon Mar 31 09:55:31 2008
 #
 CONFIG_ARM=y
 CONFIG_SYS_SUPPORTS_APM_EMULATION=y
@@ -546,7 +547,6 @@ CONFIG_SCSI_LOWLEVEL=y
 CONFIG_ATA=m
 # CONFIG_ATA_NONSTANDARD is not set
 # CONFIG_PATA_LEGACY is not set
-# CONFIG_PATA_PCMCIA is not set
 # CONFIG_PATA_QDI is not set
 # CONFIG_PATA_WINBOND_VLB is not set
 # CONFIG_PATA_PLATFORM is not set
@@ -665,6 +665,7 @@ CONFIG_VT_CONSOLE=y
 CONFIG_HW_CONSOLE=y
 # CONFIG_VT_HW_CONSOLE_BINDING is not set
 # CONFIG_SERIAL_NONSTANDARD is not set
+# CONFIG_FM_SI4702 is not set
 
 #
 # Serial drivers
@@ -1272,6 +1273,7 @@ CONFIG_ENABLE_MUST_CHECK=y
 # CONFIG_DEBUG_FS is not set
 # CONFIG_HEADERS_CHECK is not set
 # CONFIG_DEBUG_KERNEL is not set
+# CONFIG_SLUB_DEBUG_ON is not set
 # CONFIG_DEBUG_BUGVERBOSE is not set
 CONFIG_FRAME_POINTER=y
 # CONFIG_SAMPLES is not set
diff --git a/arch/arm/mach-mx3/mx3_3stack.c b/arch/arm/mach-mx3/mx3_3stack.c
index bc9098f..5a5f78e 100644
--- a/arch/arm/mach-mx3/mx3_3stack.c
+++ b/arch/arm/mach-mx3/mx3_3stack.c
@@ -804,9 +804,6 @@ static void __init mxc_board_init(void)
 
 	i2c_register_board_info(0, mxc_i2c_board_info,
 				ARRAY_SIZE(mxc_i2c_board_info));
-
-	i2c_register_board_info(0, mxc_i2c_board_info,
-				ARRAY_SIZE(mxc_i2c_board_info));
 	spi_register_board_info(mxc_spi_board_info,
 				ARRAY_SIZE(mxc_spi_board_info));
 
diff --git a/drivers/char/Kconfig b/drivers/char/Kconfig
index 29bfdbb..529bf42 100644
--- a/drivers/char/Kconfig
+++ b/drivers/char/Kconfig
@@ -413,6 +413,8 @@ config SGI_MBCS
 
 config FM_SI4702
 	tristate "SI4702 FM device driver"
+	depends on (MACH_MX31_3DS || MACH_MX35_3DS || MACH_MX37_3DS)
+	default n
 
 config MXC_MU
 	bool "MXC Messaging Unit Driver"
diff --git a/drivers/mxc/pmic/mc13783/pmic_battery.c b/drivers/mxc/pmic/mc13783/pmic_battery.c
index b3d737e..052a7b2 100644
--- a/drivers/mxc/pmic/mc13783/pmic_battery.c
+++ b/drivers/mxc/pmic/mc13783/pmic_battery.c
@@ -35,7 +35,7 @@
 #include "pmic_battery_defs.h"
 
 #include <asm/arch/pmic_power.h>
-#ifdef MXC_HWEVENT
+#ifdef CONFIG_MXC_HWEVENT
 #include <../drivers/mxc/hw_event/mxc_hw_event.h>
 #endif
 
@@ -84,7 +84,7 @@ static int open_count;		/* open count for device file */
  */
 static void callback_chg_detect(void)
 {
-#ifdef MXC_HWEVENT
+#ifdef CONFIG_MXC_HWEVENT
 	t_sensor_bits sensor;
 	struct mxc_hw_event event = { HWE_BAT_CHARGER_PLUG, 0 };
 
@@ -106,7 +106,7 @@ static void callback_chg_detect(void)
 
 static void callback_low_battery(void)
 {
-#ifdef MXC_HWEVENT
+#ifdef CONFIG_MXC_HWEVENT
 	struct mxc_hw_event event = { HWE_BAT_BATTERY_LOW, 0 };
 
 	pr_debug("In callback_low_battery\n");
@@ -117,7 +117,7 @@ static void callback_low_battery(void)
 
 static void callback_power_fail(void)
 {
-#ifdef MXC_HWEVENT
+#ifdef CONFIG_MXC_HWEVENT
 	struct mxc_hw_event event = { HWE_BAT_POWER_FAILED, 0 };
 
 	pr_debug("In callback_power_fail\n");
@@ -128,7 +128,7 @@ static void callback_power_fail(void)
 
 static void callback_chg_overvoltage(void)
 {
-#ifdef MXC_HWEVENT
+#ifdef CONFIG_MXC_HWEVENT
 	struct mxc_hw_event event = { HWE_BAT_CHARGER_OVERVOLTAGE, 0 };
 
 	pr_debug("In callback_chg_overvoltage\n");
@@ -139,7 +139,7 @@ static void callback_chg_overvoltage(void)
 
 static void callback_chg_full(void)
 {
-#ifdef MXC_HWEVENT
+#ifdef CONFIG_MXC_HWEVENT
 	t_sensor_bits sensor;
 	struct mxc_hw_event event = { HWE_BAT_CHARGER_FULL, 0 };
 
-- 
1.5.4.4

